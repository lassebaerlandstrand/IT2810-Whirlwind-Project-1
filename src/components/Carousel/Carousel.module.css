.link {
  display: flex;
  flex-direction: column;
  color: var(--primary-color);
  background-color: var(--secondary-color);
  border: none;
  border-radius: 0.5rem;
  padding: 0.5rem 1rem;
  cursor: pointer;
  font-size: 1rem;
  text-align: center;
  text-decoration: none;
  min-width: 9rem;
  box-sizing: border-box;
  align-items: center;

  h3 {
    margin: 0;
    font-size: 0.9rem;
  }

  svg {
    transition: transform 0.2s;
    width: 4rem;
    height: 4rem;
  }

  &.leftLink:hover svg {
    transform: translateX(-0.2rem);
  }

  &.rightLink:hover svg {
    transform: translateX(0.2rem);
  }
}

.buttonContainer {
  display: flex;
  justify-content: center;
  gap: 2rem;
  width: 100%;
}

.additionalInfo {
  flex-wrap: wrap;
  display: flex;
}

.svg {
  width: 100%;
  height: 100%;
}

/* Cannot use CSS variables in media query. Therefore have to specify --breakpoint-md explicitly */
@media screen and (min-width: 768px) {
  .buttonContainer {
    justify-content: space-around;
  }
}
