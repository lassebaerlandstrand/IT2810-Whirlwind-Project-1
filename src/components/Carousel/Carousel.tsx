import { IconArrowLeft, IconArrowRight } from '@tabler/icons-react';
import { Link } from 'react-router-dom';
import { WeatherInfo } from '../../types/api-types';
import LOCATIONS from '../../utils/locations';
import WeatherCard from '../WeatherCard/WeatherCard';
import styles from './Carousel.module.css';

const getPreviousCity = (currentIndex: number) => {
  if (currentIndex === 0) {
    return LOCATIONS[LOCATIONS.length - 1];
  }
  return LOCATIONS[currentIndex - 1];
};

const getNextCity = (currentIndex: number) => {
  if (currentIndex === LOCATIONS.length - 1) {
    return LOCATIONS[0];
  }
  return LOCATIONS[currentIndex + 1];
};

interface CarouselProps {
  currentIndex: number;
  data: WeatherInfo | null;
}

const Carousel: React.FC<CarouselProps> = ({ currentIndex, data }) => {
  return (
    <>
      <WeatherCard location={LOCATIONS[currentIndex]} data={data} />
      <div className={styles.buttonContainer}>
        <Link className={styles.a} to={'/location/' + getPreviousCity(currentIndex).city_name}>
          <IconArrowLeft className={styles.svg} />
          {getPreviousCity(currentIndex).city_name}
        </Link>
        <Link className={styles.a} to={'/location/' + getNextCity(currentIndex).city_name}>
          <IconArrowRight className={styles.svg} />
          {getNextCity(currentIndex).city_name}
        </Link>
      </div>
    </>
  );
};

export default Carousel;
