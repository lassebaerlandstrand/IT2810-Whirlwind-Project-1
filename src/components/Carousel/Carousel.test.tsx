import { render } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import LOCATIONS from '../../utils/locations';
import Carousel from './Carousel';

describe('Carousel', () => {
  test('should match snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/location/' + LOCATIONS[0].city_name]}>
        <Routes>
          <Route
            path="/location/:locationName"
            element={
              <Carousel
                locations={LOCATIONS}
                currentIndex={0}
                data={{
                  air_pressure_at_sea_level: 0,
                  air_temperature: 25,
                  cloud_area_fraction: 0.3,
                  relative_humidity: 0,
                  wind_from_direction: 0,
                  wind_speed: 0,
                  precipitation_amount: 0,
                }}
              />
            }
          />
        </Routes>
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
