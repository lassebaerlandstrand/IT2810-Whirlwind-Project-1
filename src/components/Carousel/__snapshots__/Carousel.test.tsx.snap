// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Carousel > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="_card_8ab24e"
  >
    <button
      aria-label="Add New York to favorites"
      aria-pressed="false"
      class="_favoriteIcon_8ab24e"
    >
      <svg
        aria-hidden="true"
        class="tabler-icon tabler-icon-heart "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5 12.572l-7.5 7.428l-7.5 -7.428a5 5 0 1 1 7.5 -6.566a5 5 0 1 1 7.5 6.572"
        />
      </svg>
    </button>
    <figure
      class="_iconContainer_8ab24e"
    >
      <svg
        aria-hidden="true"
        class="tabler-icon tabler-icon-sun-filled _svg_8ab24e"
        fill="currentColor"
        height="24"
        stroke="none"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 19a1 1 0 0 1 .993 .883l.007 .117v1a1 1 0 0 1 -1.993 .117l-.007 -.117v-1a1 1 0 0 1 1 -1z"
        />
        <path
          d="M18.313 16.91l.094 .083l.7 .7a1 1 0 0 1 -1.32 1.497l-.094 -.083l-.7 -.7a1 1 0 0 1 1.218 -1.567l.102 .07z"
        />
        <path
          d="M7.007 16.993a1 1 0 0 1 .083 1.32l-.083 .094l-.7 .7a1 1 0 0 1 -1.497 -1.32l.083 -.094l.7 -.7a1 1 0 0 1 1.414 0z"
        />
        <path
          d="M4 11a1 1 0 0 1 .117 1.993l-.117 .007h-1a1 1 0 0 1 -.117 -1.993l.117 -.007h1z"
        />
        <path
          d="M21 11a1 1 0 0 1 .117 1.993l-.117 .007h-1a1 1 0 0 1 -.117 -1.993l.117 -.007h1z"
        />
        <path
          d="M6.213 4.81l.094 .083l.7 .7a1 1 0 0 1 -1.32 1.497l-.094 -.083l-.7 -.7a1 1 0 0 1 1.217 -1.567l.102 .07z"
        />
        <path
          d="M19.107 4.893a1 1 0 0 1 .083 1.32l-.083 .094l-.7 .7a1 1 0 0 1 -1.497 -1.32l.083 -.094l.7 -.7a1 1 0 0 1 1.414 0z"
        />
        <path
          d="M12 2a1 1 0 0 1 .993 .883l.007 .117v1a1 1 0 0 1 -1.993 .117l-.007 -.117v-1a1 1 0 0 1 1 -1z"
        />
        <path
          d="M12 7a5 5 0 1 1 -4.995 5.217l-.005 -.217l.005 -.217a5 5 0 0 1 4.995 -4.783z"
        />
      </svg>
    </figure>
    <div
      class="_informationContainer_8ab24e"
    >
      <div
        class="_locationContainer_8ab24e"
      >
        <h3>
          New York
        </h3>
        <span
          aria-hidden="true"
          class="_commaSeparator_8ab24e"
        >
          , 
        </span>
        <h4>
          USA
        </h4>
      </div>
      <p
        aria-label="Temperature in New York: 25"
        class="_temperature_8ab24e"
      >
        25Â°C
      </p>
      <p>
        Clear
      </p>
    </div>
  </div>
  <div
    class="_buttonContainer_cc4f54"
  >
    <a
      aria-label="Previous city: Kringla"
      class="_link_cc4f54 _leftLink_cc4f54"
      href="/location/Kringla"
    >
      <svg
        class="tabler-icon tabler-icon-arrow-left _svg_cc4f54"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 12l14 0"
        />
        <path
          d="M5 12l6 6"
        />
        <path
          d="M5 12l6 -6"
        />
      </svg>
      <h3>
        Kringla
      </h3>
    </a>
    <a
      aria-label="Next city: Stavanger"
      class="_link_cc4f54 _rightLink_cc4f54"
      href="/location/Stavanger"
    >
      <svg
        class="tabler-icon tabler-icon-arrow-right _svg_cc4f54"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 12l14 0"
        />
        <path
          d="M13 18l6 -6"
        />
        <path
          d="M13 6l6 6"
        />
      </svg>
      <h3>
        Stavanger
      </h3>
    </a>
  </div>
</DocumentFragment>
`;
